Move External Identities APIs to v1.0 (#12210)
microsoftgraph/microsoft-graph-docs: api-reference/v1.0/resources/pkcs12certificate.md

---
title: "pkcs12Certificate resource type"

> description: "Represents configuration for uploading a pkcs12Certificate in an API call."

Reviewer: ```suggestion
description: "Represents the configuration for uploading a pkcs12Certificate."
```

author: "nickgmicrosoft"
localization_priority: Normal
ms.prod: "identity-and-sign-in"
doc_type: resourcePageType
---

# pkcs12Certificate resource type

Namespace: microsoft.graph

Represents the configuration used to **upload** a certificate when using HTTPS client-certificate authentication for calling an API connector endpoint. Client certificate authentication is a mutual certificate-based authentication, where the client provides a client certificate to an API endpoint to prove its identity. The configured certificate of an API connector is sent by Azure AD to the given API endpoint, which then validates the certificate.

Inherits from [apiAuthenticationConfigurationBase](../resources/apiauthenticationconfigurationbase.md).

## Properties

|Property|Type|Description|
|:---|:---|:---|
|pkcs12Value|String| This is the field for sending pfx content. The value should be a base-64 encoded version of the actual certificate content. Required.|

> |password|String| This is the password for the pfx file. Required. If no password is used, must still provide a value of `""`.|

Reviewer: ```suggestion
|password|String| The password for the pfx file. Required. If no password is used, you must still provide a value of `""`.|
```
